documentation/library_guide/parallel_api/pass_data_algorithms.rst

```diff
@@ -1,20 +1,36 @@
 Pass Data to Algorithms
 #######################
 
-When using the C++ standard execution policies, oneDPL supports data being passed to its algorithms as specified
-in the ISO/IEC 14882:2017 standard (commonly called C++17). According to the standard, the calling code
-must prevent data races when using algorithms with parallel execution policies.
+When using the C++ standard aligned (or *host*) execution policies, oneDPL supports data being passed to its
+algorithms as specified in the ISO/IEC 14882:2017 standard (commonly called C++17). According to the standard,
+the calling code must prevent data races when using algorithms with parallel execution policies.
 
-Note: Implementations of ``std::vector<bool>`` are not required to avoid data races for concurrent modifications
-of vector elements. Some implementations may optimize multiple ``bool`` elements into a bitfield, making it unsafe
-for multithreading. For this reason, it is recommended to avoid ``std::vector<bool>`` for anything but a read-only
-input with the C++ standard execution policies.
+.. note::
+   Implementations of ``std::vector<bool>`` are not required to avoid data races for concurrent modifications
+   of vector elements. Some implementations may optimize multiple ``bool`` elements into a bitfield, making it unsafe
+   for multithreading. For this reason, it is recommended to avoid ``std::vector<bool>`` for anything but a read-only
+   input with the C++ standard aligned execution policies.
 
 When using a device execution policy, you can use one of the following ways to pass data to an algorithm:
 
-* ``oneapi::dpl::begin`` and ``oneapi::dpl::end`` functions
-* Unified shared memory (USM) pointers
-* ``std::vector`` with or without a USM allocator
+* ``oneapi::dpl::begin`` and ``oneapi::dpl::end`` functions for a `SYCL buffer
+  <https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#subsec:buffers>`_;
+* `Unified shared memory (USM) <https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:usm>`_ pointers;
+* ``std::vector`` with or without a USM allocator.
```

Comment:
`std::vector with or without a USM allocator.`
It is not quite correct and  not consistent with the first and the second bullets  above.
Here we are telling  about  Iterator based API, and the first and the second bullet tells about first and last iterators (or pointers).
So, the third bullet should tell the same as the first I think: 
`std::begin and std::end functions for a `std::vector with or without a USM allocator.``
(Or even `oneapi::dpl::begin` `oneapi::dpl::end` for USM std::vector or we already have such customization)

Reply:
We need to be careful what we say here, we had a discussion about this when we last changed this page and chose this wording specifically because we don't want to tell people to use iterators to `std::vector` with a usm allocator.  This is explained in the final paragraph of this page.  
Some implementations of STL do not include the allocator information in the iterator, so it may trigger an additional copy to use the iterators rather than `std::vector::data()` and `std::vector::size()`.

Reply:
I reworked this page quite significantly for the ranges doc patch, please take a look.